src/Magento/FunctionalTestingFramework/StaticCheck/TestDependencyCheck.php

```diff
@@ -0,0 +1,385 @@
+                $modulePath = dirname(dirname(dirname(dirname($file))));
+                $fullModuleName = array_search($modulePath, $this->moduleNameToPath);
+                $composerModuleName = $this->moduleNameToComposerName[$fullModuleName];
+                $filenames[$item->getName()][] = $composerModuleName;
+            }
+        }
+        return $filenames;
+    }
+
+    /**
+     * Builds list of all XML files in given modulePaths + path given
+     * @param string $modulePaths
+     * @param string $path
+     * @return Finder
+     */
+    private function buildFileList($modulePaths, $path)
+    {
+        $finder = new Finder();
+        foreach ($modulePaths as $modulePath) {
+            if (!realpath($modulePath . $path)) {
+                continue;
+            }
+            $finder->files()->in($modulePath . $path)->name("*.xml");
+        }
+        return $finder->files();
+    }
+
+    /**
+     * Attempts to find any MFTF entity by its name. Returns null if none are found.
+     * @param string $name
+     * @return mixed
+     * @throws \Magento\FunctionalTestingFramework\Exceptions\TestReferenceException
+     * @throws \Magento\FunctionalTestingFramework\Exceptions\XmlException
+     */
+    private function findEntity($name)
+    {
+        if ($name == '_ENV' || $name == '_CREDS') {
+            return null;
+        }
+
+        if (DataObjectHandler::getInstance()->getObject($name)) {
+            return DataObjectHandler::getInstance()->getObject($name);
+        } elseif (PageObjectHandler::getInstance()->getObject($name)) {
+            return PageObjectHandler::getInstance()->getObject($name);
+        } elseif (SectionObjectHandler::getInstance()->getObject($name)) {
+            return SectionObjectHandler::getInstance()->getObject($name);
+        }
+
+        try {
+            return ActionGroupObjectHandler::getInstance()->getObject($name);
+        } catch (TestReferenceException $e) {
+        }
+        try {
+            return TestObjectHandler::getInstance()->getObject($name);
+        } catch (TestReferenceException $e) {
+        }
+        return null;
+    }
+
+    /**
+     * Prints out given errors to file, and returns summary result string
+     * @param array $errors
+     * @return string
+     */
+    private function printErrorsToFile($errors)
+    {
+        if (empty($errors)) {
+            return "No Dependency errors found.";
+        }
+        $outputPath = getcwd() . DIRECTORY_SEPARATOR . "mftf-dependency-checks.txt";
+        $fileResource = fopen($outputPath, 'w');
+        $header = "MFTF File Dependency Check:\n";
+        fwrite($fileResource, $header);
+        foreach ($errors as $test => $error) {
+            fwrite($fileResource, $error[0] . PHP_EOL);
+        }
+        fclose($fileResource);
+        $errorCount = count($errors);
+        $output = "Dependency errors found across {$errorCount} file(s). Error details output to {$outputPath}";
```

Comment:
Is it number total tests scanned or number of total errors found? Just look at the error message in console confused me until I see the source code.

Reply:
Errors unique to file, the `$error` array is key => value of `fileName => [$errors]` so the count just counts unique file names.